test/test.sh

```diff
@@ -1,30 +1,35 @@
-#!/bin/bash -e
+#!/bin/sh
+
+set -eux
 
 export TZ='US/Pacific'
 
-for pcap in `ls test/pcap`
-do
+for pcap in test/pcap/*.pcap; do
     TMP=$(mktemp -d)
-    if [ -z "${TMP}" ]; then
+    if [ ! -d "${TMP}" ]; then
         echo "Failed to create temporary directory '${TMP}'"
         break
     fi
     TMPSE2CSV="${TMP}/se2csv"
-
-    SAMPLE="${pcap%.*}"
+    SAMPLE="$(basename --suffix=".pcap" "${pcap}")"
+    REC_OPTION="-r ${TMP}/${SAMPLE}.rec"
+    OUT_OPTION="-o ${TMP}/${SAMPLE}.json"
     KEY_OPTION=""
-    if [ -f "test/keys/${SAMPLE}.key" ]
-    then
+    if [ -f "test/keys/${SAMPLE}.key" ]; then
         KEY_OPTION="-k test/keys/${SAMPLE}.key"
     fi
-    echo "Running test sample: ${SAMPLE}"
-    echo "Key option: ${KEY_OPTION}"
-    tshark -r "test/pcap/${SAMPLE}.pcap" -T fields -e data | ./utilities/unhexlify.py | ./semonitor.py -r "${TMP}/${SAMPLE}.rec" -o "${TMP}/${SAMPLE}.json" - $KEY_OPTION
+    SEMONITOR_OPTIONS="${REC_OPTION} ${OUT_OPTION} - ${KEY_OPTION}"
+
+    tshark -r "test/pcap/${SAMPLE}.pcap" -T fields -e data | ./utilities/unhexlify.py | eval ./semonitor.py "${SEMONITOR_OPTIONS}"
```

Comment:
why do you need the `eval`? if you remove the quotes around `${SEMONITOR_OPTIONS}` it should expand as expected

Reply:
also, whats the value in separating out these options into a variable if it isn't reused?  i think it is more easily understood if the options to semonitor are inline

Reply:
if you remove the quotes, you run into linting issues :)
SEMONITOR_OPTIONS=${REC_OPTION} ${OUT_OPTION} - ${KEY_OPTION}
                                                ^-- SC2086: Double quote to prevent globbing and word splitting.
and iirc SC2086 thus then recommends either using arrays or eval.

As for having them split up; They will be re-used in the other PR/future proofing.

Reply:
well you either use eval and quite the variable (eval's purpose sort of is to 'build a string with arguments, and then send it to eval for execution) or use unquoted strings. I know we shouldn't change code to please a linter, but it does make sense.